### ISSUE TITLE
Suggestion: using `aria` application role for the sonified charts

### ISSUE THREAD

**Issue by jooyoungseo** [julianna-langston/chart2music#20]
Sonified image elements by c2m require screen readers' focus mode/forms mode. Do you have any special reasons that you did not implement "application" role for the users not having to manually switch between modes? I have also noticed that SAS Graphic Accelerator used "application" role.

The following example automatically switches to screen readers' focus/forms mode when it gets focused via tab key.

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>

    <script src="https://cdn.jsdelivr.net/npm/chart2music"></script>
</head>

<body>

    <img src="mychart.png" id="MyChart" role="application"/>

    <script>
        c2mChart({
            type: "bar",
            element: document.getElementById("MyChart"),
            data: [1, 2, 3]
        });
    </script>
</body>

</html>
```

**Comment by julianna-ciq** [comment]
Sorry for not responding, I just want to update you on the status of this.

By default, we assign `role=img` to target elements (that aren't already image tags). We did this to get Firefox to play nicely with the target element. 
We're looking into using `role=application`, and confirm that it performs well with user testing. We'll update you with what we find in the next few days

**Comment by julianna-ciq** [comment]
We are evaluating this change with this PR: https://github.com/julianna-langston/chart2music/pull/26

**Comment by julianna-ciq** [comment]
We completed evaluations, merged in PR #26, and have published a new version (1.3.2).

@jooyoungseo , can you test to see if v1.3.2 fits your needs? If not, please let me know if there's anything else we can do.

**Comment by jooyoungseo** [comment]
@julianna-ciq  -- Thanks for the update. After #26, the "application" role works nicely. However, There is an unexpected side-effect. An `img` element modified by c2m is no longer recognizable by screen readers when navigated via up/down-arrows. The only way to recognize there is an interactive image is via tab key access. In sum, screen readers cannot recognize the existence of the image in virtual cursor (in JAWS, for example) when navigating a page with up/down arrows. I am sorry, but I think the prior behavior was safer and reliable for a wider screen reader audience unless this issue cannot be easily addressed.

**Comment by jooyoungseo** [comment]
I have tested with other screen reader, NVDA. Actually, this is a tradeoff. Since we can only assign one aria role per element, we will lose a way to let screen readers know the element as an image if we use "application." I think this is why you kept img role due to this issue. I am sorry for confusing your team. I think we may want to keep the prior behavior and just instruct screen reader users to use forms/focus mode when interacting with the chart. I think that's much safer....

**Comment by spmealin** [comment]
I think it’s OK with roll=application since screen readers say “sonified graph” when you browse past or tab to a chart.

Sean (Sent from my smartphone)

**Comment by jooyoungseo** [comment]
The primary issue is that image elements are not semantically recognized as graphics. Say, for example, if there are multiple images on a page and screen reader users may use jump between graphic elements via single-letter quick navigation. Or, sometimes we pull up a list of graphics contained in a page. That kind of nav is unavailable with application role. Another issue is that alt attribute is not read for "application" role elements for some screen readers when accessed via virtual cursor up and down arrow keys. If you would like to keep the "application" role, you may want to consider using `aria-label` instead of `alt` attribute to address  that issue.

**Comment by jooyoungseo** [comment]
I have a better idea. @julianna-ciq -- Do you think it's possible to add onclick event to image elements? You may be able to change the role value to "appication" against onclick event.

**Comment by julianna-ciq** [comment]
@jooyoungseo , I think that `role=application` is a more generalizable role to apply. That said, if a programmer or report author wants to override that role, they can simply assign a role of their own. I would prefer to leave the role by default, and defer to the report author to override if they deem it necessary.

I have added a new page to the Chart2Music documentation site that elaborates on how C2M uses the element, as well as the pros and cons of different roles an author could assign.

Since we can only choose one role, and they both have their pluses and minuses, this is what I feel is the best balance to strike. How do you feel about this compromise?

**Comment by jooyoungseo** [comment]
@julianna-ciq -- I think that's great! Yes, each role has their own pros and cons. Users may choose between the tradeoffs. Please feel free to close this issue anytime. :)
